### PR TITLE
Specify header sizes in style.css and remove bold formatting from contributing.md headers

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**
This PR specifies the header styles in style.css based on the comments in https://github.com/GenericMappingTools/pygmt/pull/1339#discussion_r659132348 and https://github.com/GenericMappingTools/pygmt/pull/1335#pullrequestreview-685712272 that header 4 levels are too small. The header sizes decrease linearly from 200% for header 1.

For explanation, in https://github.com/GenericMappingTools/pygmt/pull/1335# I suggested that this was a MyST issue, but in testing a new issue for conversion from .md to .rst I discovered that the ReST versus Markdown format did not affect the header appearance, hence the fix in style.css and the removal of the temporary fix in bolding the headers in contributing.md.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: doc/_static/style.css
Comment (weiji14) [inline doc/_static/style.css:30]: Check with others if the 20% difference between each step is large enough. A quick web search seems to suggest that browsers use a default scaling of h1: 200%, h2: 150%, h3: 117%, h4: 100% (rather large steps). Did you try other increments?

**Review by weiji14 — APPROVED** [review]
Looks much better! I'd wait for a second (or second) opinion/approval before merging directly, preferably someone on a different browser (mine is Firefox).

File: doc/_static/style.css
Comment (meghanrjones) [reply doc/_static/style.css:30]: Yeah, I tried a few options but nothing super systematic. From trial and error, it seemed that 130% for h4 is the minimum that makes it obvious that it's a header. IMO h1 already looked good at 200%. So we could increase the spacing to from 130% to 200% or increase the maximum size past 200% to increase step size.

Comment (core-man) [reply doc/_static/style.css:30]: Do you think it's better to have a larger step between h2 and h3, which may be commonly used? h1 is only used once and h4 is used a few times for each chapter. e.g., h2: 185%, h3: 150%, h4: 130%?

See what I see on Chrome below. It seems that **Providing Feedback** (h2) is not well separated from its h3 subtitles (e.g., **Reporting a Bug**).

Comment (meghanrjones) [reply doc/_static/style.css]: Good idea, implemented in https://github.com/GenericMappingTools/pygmt/pull/1385/commits/1fbb9061068ac8f28a21bc4d146c3dfa78f81152.

**Review by core-man — APPROVED** [review]
Looks good on Chrome on a 22-inch display and Safari on a 13 inch one~

**Review by willschlitzer — APPROVED** [review]
Looks good to me (Ubuntu, Firefox, 21 inch monitor)

**Pull request merged**
